hoomd/hpmc/PairPotentialUnion.cc

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#include "PairPotentialUnion.h"
+
+#include "hoomd/hpmc/OBBTree.h"
+
+namespace hoomd
+    {
+namespace hpmc
+    {
+PairPotentialUnion::PairPotentialUnion(std::shared_ptr<SystemDefinition> sysdef,
+                                       std::shared_ptr<PairPotential> constituent_potential)
+    : PairPotential(sysdef), m_constituent_potential(constituent_potential), m_leaf_capacity(4)
+    {
+    unsigned int ntypes = m_sysdef->getParticleData()->getNTypes();
+    m_extent_type.resize(ntypes, 0.0);
+    m_type.resize(ntypes);
+    m_position.resize(ntypes);
+    m_orientation.resize(ntypes);
+    m_charge.resize(ntypes);
+    m_tree.resize(ntypes);
+    }
+
+void PairPotentialUnion::setBody(std::string body_type, pybind11::object v)
+    {
+    unsigned int body_type_id = m_sysdef->getParticleData()->getTypeByName(body_type);
+
+    if (v.is_none())
+        {
+        m_extent_type[body_type_id] = 0;
+        m_position[body_type_id].clear();
+        m_orientation[body_type_id].clear();
+        m_charge[body_type_id].clear();
+        m_type[body_type_id].clear();
+        return;
+        }
+
+    pybind11::list positions = v["positions"];
+    pybind11::object types_obj = v["types"];
+    pybind11::object orientations;
+    pybind11::object charges;
+    if (v.contains("orientations"))
+        {
+        orientations = v["orientations"];
+        }
+    else
+        {
+        orientations = pybind11::none();
+        }
+    if (v.contains("charges"))
+        {
+        charges = v["charges"];
+        }
+    else
+        {
+        charges = pybind11::none();
+        }
```

Comment:
```suggestion
    pybind11::object orientations = v.contains("orientations") ? v["orientations"] : pybind11:none();
    pybind11::object charges = v.contains("charges") ? v["charges"] : pybind11::none();
```
Good candidate for the ternary operator.

Reply:
This change produces the error: `TypeError: Object of type 'list' is not an instance of 'none'`. I am reverting.